core/primitives/src/utils.rs

```diff
@@ -180,6 +180,16 @@ pub fn is_valid_sub_account_id(signer_id: &AccountId, sub_account_id: &AccountId
     VALID_ACCOUNT_PART_ID_WITH_TAIL_SEPARATOR.is_match(prefix)
 }
 
+/// Returns a value of the median index of the array, NOT a median value.
+pub fn median<T: std::clone::Clone + std::cmp::Ord>(arr: Vec<T>) -> T {
+    if arr.is_empty() {
+        panic!("empty array")
+    }
+    let mut arr = arr.clone();
+    arr.sort();
+    arr[arr.len() / 2].clone()
+}
+
```

Comment:
What is the difference between this and the median (the only I see is when size is even that you pick the bigger of the two in the middle) but other than that it looks like the median to me.

Two notes about performance:

1. It can be implemented in [linear time (no need to sort)](https://en.wikipedia.org/wiki/Median_of_medians). Haven't used this algorithm before, so I'm not sure if it is really better for the size of arrays we are aiming at (I'm also not sure about what this size is).

2. You can avoid cloning big objects by passing only reference to the objects and finding median using vector of index.

AFAISee from the code, this is only used for vector of integers, and if the length is "short" then this is definitely the best implementation regarding both items, since integer types are Copy.

Reply:
> What is the difference between this and the median (the only I see is when size is even that you pick the bigger of the two in the middle) but other than that it looks like the median to me.

That's correct, I'm not calculating mean of two middle values. It works for our particular purpose well. Anyway, I think it's important to say about it in comment.

> Two notes about performance:

You are right, there is linear algorithm exists. As we only use median for vector of block producers, we may not go with further optimizations at this point.